analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/SymbolicExecutionRunnerTest.cs

```diff
@@ -395,6 +395,10 @@ public void Method()
                 .StartWith("Analyzer 'SonarAnalyzer.UnitTest.Rules.SymbolicExecution.SymbolicExecutionRunnerTest+ConfigurableSERunnerCS' threw an exception of type 'SonarAnalyzer.SymbolicExecution.SymbolicExecutionException' with message 'Error processing method: Method ## Method file: snippet1.cs ## Method line: 4,4 ## Inner exception: System.InvalidOperationException: This check is not useful. ##    at SonarAnalyzer.UnitTest.TestFramework.SymbolicExecution.ThrowAssignmentRuleCheck.PostProcessSimple(SymbolicContext context)");
         }
 
+        [TestMethod]
+        public void Analyze_DeconstructionOfNestedTuple() =>
+            Verify(@"var (_, (_, _)) = (1, (2, 3));", ProjectType.Product, null);
```

Comment:
If it helps, my real world use case (Which was reduced to the above to reproduce the issue) was indeed `var (a, (b, c)) = `
Could you have a parameterised test that tries all the variants?  The workarounds you mentioned in the community forums + the actual issue and a couple of variations of same?
It seems like your `c = 3` case is a different problem to solve?

Reply:
This kind of test should go to `SonarExplodedGraphTest.cs`, but I'm not sure it's reasonably easy there to test it.  In case it turns out to be complicated, you can add it to any SE-based rule. I'd choose S2583 / S2589 as you can document some FNs there with the case.